lib/jekyll/collection.rb

```diff
@@ -204,5 +195,24 @@ def extract_metadata
         {}
       end
     end
+
+    private
+    def read_docs(full_path)
+      doc = Jekyll::Document.new(full_path, { :site => site, :collection => self })
+      doc.read
+      if site.publisher.publish?(doc) || !write?
+        docs << doc
+      else
+        Jekyll.logger.debug "Skipped From Publishing:", doc.relative_path
+      end
+    end
+
+    private
+    def read_files(file_path, full_path)
+      relative_dir = Jekyll.sanitized_path(relative_directory,
+                                           File.dirname(file_path)).chomp("/.")
```

Comment:
This is very uncleanly and offsets the eyes IMO.  Why not just fold the `()` and put the data between?